### PR TITLE
Add support for specifying table name in show shadow table rule for shadow table feature

### LINKED ISSUE #25925: Add `SHOW SHADOW TABLE RULE ruleName` for shadow table.
### Is your feature request related to a problem?
No.

### Describe the feature you would like.
Add `SHOW SHADOW TABLE RULE ruleName` for shadow table.

FYI: 
- [SHOW SHADOW TABLE RULE](https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-proxy/distsql/syntax/rql/rule-query/shadow/show-shadow-table-rule/)
- [SHOW SHADOW RULE](https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-proxy/distsql/syntax/rql/rule-query/shadow/show-shadow-rule/)

### PR DESCRIPTION
**Pull request by HarshSawarkar**
Fixes #25925 .

Changes proposed in this pull request:
  -
Added support for specifying table name to the feature Show Shadow Table Rules. 
---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by RaigorJiang** [comment]
Hi @HarshSawarkar 
Thank you for submitting this PR, but sorry, I did not see the changes of g4 files, which means there is no new syntax support.
Have you executed DistSQL to verify it?

**Comment by HarshSawarkar** [comment]
> Hi @HarshSawarkar Thank you for submitting this PR, but sorry, I did not see the changes of g4 files, which means there is no new syntax support. Have you executed DistSQL to verify it?

Thanks for your response.I haven't made changes in g4 files.Can you please help me with how to proceed further and where to make changes.And also how do i execute DistSQL module?

**Comment by HarshSawarkar** [comment]
> > Hi @HarshSawarkar Thank you for submitting this PR, but sorry, I did not see the changes of g4 files, which means there is no new syntax support. Have you executed DistSQL to verify it?
> 
> Thanks for your response.I haven't made changes in g4 files.Can you please help me with how to proceed further and where to make changes.And also how do i execute DistSQL module?

**Comment by RaigorJiang** [comment]
Hi @HarshSawarkar ,
For reference, there are two blogs about how to use DistSQL:
- https://shardingsphere.apache.org/blog/en/material/2022_07_26_distsql_applications_building_a_dynamic_distributed_database/
- https://shardingsphere.apache.org/blog/en/material/2023_01_04_refactoring_the_distsql_syntax__shardingsphere_5.3.0_deep_dive/

And another one guide us how to develop DistSQL:
- https://shardingsphere.apache.org/blog/en/material/jan_28_blog_x_how_to-develop_your_distributed_sql_statement_in_apache_shardingsphere/

**Comment by HarshSawarkar** [comment]
Thanks @RaigorJiang I am going through the blogs and will make the changes in the ANTLR file.

**Comment by HarshSawarkar** [comment]
Hey @RaigorJiang Can you please review the chamges now?

File: features/shadow/distsql/parser/src/main/antlr4/imports/shadow/RQLStatement.g4 [hunk 1]
```diff
@@ -43,6 +43,10 @@ countShadowRule
     : COUNT SHADOW RULE (FROM databaseName)?
     ;
 
+tableRule
+    :RULE tableName
```
Comment (RaigorJiang) [inline features/shadow/distsql/parser/src/main/antlr4/imports/shadow/RQLStatement.g4:47]: Please keep the format consistent: there is a space after the colon

File: features/shadow/distsql/parser/src/main/java/org/apache/shardingsphere/shadow/distsql/parser/core/ShadowDistSQLStatementVisitor.java [hunk 2]
```diff
@@ -152,7 +152,7 @@ public ASTNode visitShowShadowRules(final ShowShadowRulesContext ctx) {
     
     @Override
     public ASTNode visitShowShadowTableRules(final ShowShadowTableRulesContext ctx) {
-        return new ShowShadowTableRulesStatement(null == ctx.databaseName() ? null : (DatabaseSegment) visit(ctx.databaseName()));
+        return new ShowShadowTableRulesStatement(null == ctx.databaseName() ? null : (DatabaseSegment) visit(ctx.databaseName()), "");
```
Comment (RaigorJiang) [inline features/shadow/distsql/parser/src/main/java/org/apache/shardingsphere/shadow/distsql/parser/core/ShadowDistSQLStatementVisitor.java:155]: In this visitor method, we should get table name from ctx and put into statement.

**Review by RaigorJiang — CHANGES_REQUESTED** [review]

**Comment by RaigorJiang** [comment]
Hi @HarshSawarkar,
Thanks for your contribution, this PR is great!  And I'll make some test later.

**Comment by HarshSawarkar** [comment]
Thanks @RaigorJiang for your support

> Hi @HarshSawarkar, Thanks for your contribution, this PR is great! And I'll make some test later.

Thanks, @RaigorJiang , for your help and support! I'm looking forward to making more contributions. By the way, could you please add me to the ASF Slack? Here's my email ID: [harshsawarkar111@gmail.com](mailto:harshsawarkar111@gmail.com).

**Pull request merged**